js_modules/dagit/packages/core/src/assets/AssetNodeDefinition.tsx

```diff
@@ -183,23 +184,58 @@ const DefinitionLocation: React.FC<{
           />
         </Mono>
       ))}
-    {displayNameForAssetKey(assetNode.assetKey) !== assetNode.opName && assetNode.opName && (
-      <Box flex={{gap: 6, alignItems: 'center'}}>
-        <Icon name="op" size={16} />
-        <Mono>{assetNode.opName}</Mono>
-      </Box>
-    )}
-
+    <OpNamesDisplay assetNode={assetNode} repoAddress={repoAddress} />
     {isSourceAsset(assetNode) && (
       <Caption style={{lineHeight: '16px', marginTop: 2}}>Source Asset</Caption>
     )}
   </Box>
 );
 
+const OpNamesDisplay = (props: {
+  assetNode: AssetNodeDefinitionFragment;
+  repoAddress: RepoAddress;
+}) => {
+  const {assetNode, repoAddress} = props;
+  const {assetKey, graphName, opNames} = assetNode;
+  const opCount = opNames.length;
+
+  if (!opCount) {
+    return null;
+  }
+
+  const firstOp = opNames[0];
+  if (opCount === 1) {
```

Comment:
I think it might technically be more accurate to check if graph name is null. this will reliably tell you if you have a graph-backed asset or not